scrnaseq.Rmd

```diff
@@ -882,22 +847,35 @@ cat("## Integration of multiple datasets\n\n")
 ```
 
 ```{r multi_dataset_integration_merge, eval=(param$integrate_samples[["method"]]=="merge"), include=(param$integrate_samples[["method"]]=="merge")}
-# Called when no integration is needed (e.g. samples were multiplexed on the same chip)
+# Data for different samples can be merged if no integration is needed, 
+#   for example, when samples were multiplexed on the same chip
 if (param$integrate_samples[["method"]]=="merge") {
   sc = merge(x=sc[[1]], y=sc[2:length(sc)], project=param$project_id)
+
+  # Re-score cell cycle effects after merge
+  if (param$cc_rescore_after_merge) {
+    sc = cc_scoring(sc=sc, genes_s=genes_s[,2], genes_g2m=genes_g2m[,2])
+    if (all(is.na(sc[["S.Score"]])) & all(is.na(sc[["G2M.Score"]]))) param$cc_remove=FALSE
+  }
   
-  # Find variable features in RNA assay
-  DefaultAssay(sc) = "RNA"
-  sc = Seurat::FindVariableFeatures(sc, selection.method = "vst", nfeatures = 3000, verbose=FALSE)
+  # (Re-)Run normalisation, variable features and scaling
+  if (param$norm == "RNA") {
+    # Find variable features in RNA assay
+    sc = Seurat::FindVariableFeatures(sc, selection.method = "vst", nfeatures = 3000, verbose=FALSE)
    
-  # Scale RNA assay
-  # Note: Removing cell cycle effects in "RNA" scaled data can be very slow
-  sc = Seurat::ScaleData(sc, features=rownames(sc[["RNA"]]), vars.to.regress=param$vars_to_regress, verbose=FALSE, assay="RNA")
+    # Scale RNA assay
+    # Note: Removing cell cycle effects in "RNA" scaled data can be very slow
+    sc = Seurat::ScaleData(sc, features=rownames(sc[["RNA"]]), vars.to.regress=param$vars_to_regress, verbose=FALSE, assay="RNA")
   
-  # Rerun SCTransform
-  DefaultAssay(sc) = "SCT"
-  min_cells_overall = max(purrr::map_int(param$feature_filter, function(f) {as.integer(f[["min_cells"]])}))
-  sc = suppressWarnings(SCTransform(sc, vars.to.regress=param$vars_to_regress, min_cells=min_cells_overall, verbose=FALSE, return.only.var.genes=FALSE))
+  } else if (param$norm == "SCT") {
+    # Rerun SCTransform
+    min_cells_overall = max(purrr::map_int(param$feature_filter, function(f) as.integer(f[["min_cells"]])))
+    sc = SCTransform(sc, 
+                     vars.to.regress=param$vars_to_regress, 
```

Comment:
Only use vars_to_regress instead of param$vars_to_regress. Same for feature_filter.

Reply:
I don't understand this one. Why?

Reply:
Missed that this is inside a function. Solved.